optimlol_api/persistence/dataProviders/statsDataProvider.js

```diff
@@ -59,9 +63,21 @@ module.exports = function() {
 			.then(function(cacheStatsResult) {
 				if (cacheStatsResult.isExpired === false) {
 					_logger.debug("Using cached stats.");
+					var statsResult = _prepareStats(cacheStatsResult);
+					statsResult.data.quality = "fresh"
 					deferred.resolve(_prepareStats(cacheStatsResult));
 				} else {
-					_getStatsApi(region, summonerId, deferred);
+					_getStatsApi(region, summonerId)
+						.then(function(statsResult) {
+							statsResult.data.quality = "fresh";
+							deferred.resolve(statsResult);
+						})
+						.fail(function(RiotFailure){
+							_logger.debug("Using cached stats. Because Riot is down");
+							var statsResult = _prepareStats(cacheStatsResult);
+							statsResult.data.quality = "stale"
```

Comment:
I don't want to add the property onto the data itself. Lets add it to the statsResult.

```statsResult.quality```

But I do like this idea.